### PR TITLE
fix: object with numeric values in start are unusable

### PR DESCRIPTION
**Pull request by jimil749**
fix for #1557 
Added a check using isdigit to check if the first character is not numeric.

GDevelop-2

One issue is, this shows the same pop-up stating it contains forbidden characters, whereas here it does not, so should we change the pop or just modify the text in it.

### CODE REVIEW

**Comment by 4ian** [comment]
Thanks! :) Looks good. Also update `GetBadObjectNameWarning` with a concise explanation about not starting object name with a digit :)

**Comment by jimil749** [comment]
Along with updating the `GetBadObjectNameWarning`, shouldn't we also update the warning displayed to the user which is inside of the MainFrame? Because that would make users think what went wrong! So I took the liberty and changed the warning message if that's fine!

**Comment by jimil749** [comment]
Along with updating the `GetBadObjectNameWarning`, shouldn't we also update the warning shown to the user? Else they would think what went wrong! So I took the liberty to update warning message! :)

**Comment by 4ian** [comment]
Yes that's a good idea, actually we should change the warning for ALL messages in the codebase. That's a good thing to do whenever you change something: check the rest of the codebase. How? Just search for validateObjectName.

I'll do that for you :)

**Comment by jimil749** [comment]
No worries, I'll make the changes right away. Shouldn't take much time, as currently it's only updated for the object.

**Comment by 4ian** [comment]
Actually validateObjectName is ill named (my fault) and is used for quite a few other name validation => all needs to be updated.

Use `This name is invalid. Only use alphanumeric characters (0-9, a-z) and underscores. Digits are not allowed as the first character.` for the user message :)

**Comment by jimil749** [comment]
So should we change the name to validateVariableName? Makes more sense, if it is used to validate other variables?

**Comment by 4ian** [comment]
Maybe, or just "validateName" (a variable has another meaning in GDevelop)

**Comment by jimil749** [comment]
Weird that the tests are failing! I did not touch the file failing. 
Any idea why is that happening?

File: newIDE/app/src/EventsFunctionsExtensionEditor/index.js [hunk 1]
```diff
@@ -219,18 +219,18 @@ export default class EventsFunctionsExtensionEditor extends React.Component<
...
-          t`This name contains forbidden characters: please only use alphanumeric characters (0-9, a-z) and underscores in your function name.`
+          t`This name is invalid. Only use alphanumeric characters (0-9, a-z) and underscores. Digits are not allowed as the first character.`
         )
       );
       return;
     }
     if (isExtensionLifecycleEventsFunction(newName)) {
       showWarningBox(
         i18n._(
-          t`This name is reserved for a lifecycle function of the extension. Choose another name for your function.`
```
Comment (4ian) [inline newIDE/app/src/EventsFunctionsExtensionEditor/index.js]: Careful, you've changed the wrong message here, it's a regression

Comment (jimil749) [reply newIDE/app/src/EventsFunctionsExtensionEditor/index.js]: Mistake on my part. Updated the wrong message here.

Comment (4ian) [reply newIDE/app/src/EventsFunctionsExtensionEditor/index.js]: I fixed this for you, no need to push anything

**Comment by 4ian** [comment]
Merged :) Thanks for your work on this!
I'd advise to always double check your changes when you make a PR. For example, using the search function, search for the sentence that you've changed, and check that the count is making sense according to the count for the previous sentence. That's how I found that you renamed an extra sentence that should not have been.

Thanks :)  👍

**Comment by jimil749** [comment]
Thanks for merging and also I appreciate you helping me with all the advice! Will be sure to be more careful the next time. :)
Looking forward to solve more issues! ;)

**Pull request merged**